nvm.psm1

```diff
@@ -2,6 +2,20 @@
 Set-StrictMode -Version Latest
 $ErrorActionPreference = 'Stop'
 
+$completion_AvailableNodeVersions = {
+    param($commandName, $parameterName, $wordToComplete, $commandAst, $fakeBoundParameter)
+
+    Get-NodeVersions | Where-Object { $_ -like "$wordToComplete*" } | ForEach-Object {
+        New-Object System.Management.Automation.CompletionResult $_, $_, 'ParameterValue', ('{0} ({1})' -f $_, $_)
+    }
+}
+
+if (-not (Test-Path variable:global:options)) { $global:options = @{CustomArgumentCompleters = @{};NativeArgumentCompleters = @{}}}
+$global:options['CustomArgumentCompleters']['Remove-NodeVersion:Version'] = $completion_AvailableNodeVersions
+$global:options['CustomArgumentCompleters']['Set-NodeVersion:Version'] = $completion_AvailableNodeVersions
+
+$function:tabexpansion2 = $function:tabexpansion2 -replace 'End\r\n{','End { if ($null -ne $options) { $options += $global:options} else {$options = $global:options}'
```

Comment:
I don't know a lot about PowerShell, but is this really the recommended way to do it? Google search showed mixed approaches

Reply:
Yeah from googling that's the best way I've found. I don't quite get how it works or what that final line does (no one seems to explain it).